projects/com.oracle.truffle.llvm.parser/src/com/oracle/truffle/llvm/parser/LazyToTruffleConverterImpl.java

```diff
@@ -109,21 +117,83 @@ public RootCallTarget convert() {
                         method.getParameters().size(), symbols,
                         method, liveness, notNullable, dbgInfoHandler);
         method.accept(visitor);
+
+        LLVMControlFlowGraph cfg = new LLVMControlFlowGraph(method.getBlocks().toArray(new InstructionBlock[method.getBlocks().size()]));
+        try {
+            cfg.build();
+        } catch (Exception e) {
```

Comment:
catching exception is a bit too generic - this will hide errors in the implementation of the loop detection - actually, I don't see any reason why the build method should throw an exception at all (it should just return and set the isReducible flag of cfg).